src/test/scala/CounterSpecification.scala

```diff
@@ -0,0 +1,38 @@
+import Cqrs.Aggregate._
+import Cqrs.Database.EventSerialisation
+import Domain.Counter
+import org.scalacheck.{Properties, Gen}
+
+object CounterSpecification extends Properties("Counter") {
+
+  property("fails when below zero") = CounterSpec.property()
+
+  private object CounterSpec extends AggregateCommands[Counter.Event, Counter.flow.StateData, Int] {
```

Comment:
@Fristi hi! 

what do you think about this kind of test for an aggregate? I'll create a similar one for the door agg.

projections, however, will need something more clever than this...

is there an alternative approach you can think of?